pypeit/find_objects.py

```diff
@@ -1143,56 +1204,41 @@ def joint_skysub(self, skymask=None, update_crmask=True, trim_edg=(0,0),
             update_crmask = False
             if not self.par['reduce']['skysub']['global_sky_std']:
                 msgs.info('Skipping global sky-subtraction for standard star.')
-                return global_sky
-
+                return _global_sky
+
+        # Use the FWHM map determined from the arc lines to convert the science frame
+        # to have the same effective spectral resolution.
+        fwhm_map = self.wv_calib.build_fwhmimg(self.tilts, self.slits, initial=True, spat_flexure=self.spat_flexure_shift)
+        thismask = thismask & (fwhm_map != 0.0)
```

Comment:
Can you please add some documentation to this routine explaining how the joint sky subtraction works. I don't follow the philosophy here. Why you are convolving the image to a single fwhm and then fitting the global_sky. It looks like this is only used to compute the scaleimg. A detailed description of the control flow of the routine is needed in the docstring.